### PR TITLE
PAY-1411 2024 rates and constants

### PR DESCRIPTION
**Pull request by mark-h-humi**
updated rates and constants for federal and provincial tax, EI, QPP and QPIP

https://gethumi.atlassian.net/browse/PAY-1411 also https://gethumi.atlassian.net/browse/PAY-1412

https://www.canada.ca/en/revenue-agency/services/forms-publications/payroll/t4127-payroll-deductions-formulas/t4127-jan/t4127-jan-payroll-deductions-formulas-computer-programs.html#toc77 table 8.2, 8.3, 8.7 and 8.8

### CODE REVIEW

File: lib/taxman/taxman2024/nb/t4.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@ module Nb
...
+      DEFAULT_TD1 = 13_044_00.to_d
 
       RATES_AND_CONSTANTS = {
-        47_715_00.to_d => [LOWEST_RATE, 0.0.to_d],
-        95_431_00.to_d => [0.1400, 2_195_00.to_d],
-        176_756_00.to_d => [0.1600, 4_104_00.to_d],
+        49_958_00.to_d => [LOWEST_RATE, 0.0.to_d],
+        99_916_00.to_d => [0.1400, 2_195_00.to_d],
+        185_064_00.to_d => [0.1600, 4_104_00.to_d],
         BigDecimal("Infinity") => [0.1950, 10_290_00.to_d]
```
Comment (claus-humi) [inline lib/taxman/taxman2024/nb/t4.rb:14]:
```suggestion
        BigDecimal("Infinity") => [0.1950, 10_774_00.to_d]
```

**Review by claus-humi — CHANGES_REQUESTED** [review]
No changes for NS?

**Comment by mark-h-humi** [comment]
> No changes for NS?

correct

File: lib/taxman/taxman2024/nb/t4.rb (hunk 1 above)
Comment (mark-h-humi) [reply lib/taxman/taxman2024/nb/t4.rb:14]: oops

File: lib/taxman/taxman2024/t3.rb [hunk 2]
```diff
@@ -6,11 +6,11 @@ class T3 < Factor
...
-      53_359_00.to_d => [LOWEST_RATE, 0.0.to_d],
-      106_717_00.to_d => [0.205.to_d, 2_935_00.to_d],
-      165_430_00.to_d => [0.260.to_d, 8_804_00.to_d],
-      235_675_00.to_d => [0.290.to_d, 13_767_00.to_d],
-      BigDecimal("Infinity") => [0.330.to_d, 23_194_00.to_d]
+      55_867_00.to_d => [LOWEST_RATE, 0.0.to_d],
+      111_733_00.to_d => [0.205.to_d, 3_073_00.to_d],
+      173_205_00.to_d => [0.260.to_d, 9_218_00.to_d],
+      246_752_00.to_d => [0.290.to_d, 14_414_00.to_d],
+      BigDecimal("Infinity") => [0.330.to_d, 24_284_00.to_d]
```
Comment (markjconlon-humi) [inline lib/taxman/taxman2024/t3.rb:13]: lower in this class there is a CEA method that is hardcoded to 1_368_00 does that need to be changed to 1_433_00 for 2024?

**Review by markjconlon-humi — CHANGES_REQUESTED** [review]
Couple changes needed but looking good, usually takes a few tries with all the numbers

File: lib/taxman/taxman2024/ab/t4.rb [hunk 3]
```diff
@@ -5,14 +5,14 @@ module Ab
...
-      DEFAULT_TD1 = 21_003_00.to_d
+      DEFAULT_TD1 = 21_885_00.to_d
 
       RATES_AND_CONSTANTS = {
-        142_292_00.to_d => [LOWEST_RATE, 0.0.to_d],
-        170_751_00.to_d => [0.12, 2_846_00.to_d],
-        227_668_00.to_d => [0.13, 4_553_00.to_d],
-        341_502_00.to_d => [0.14, 6_830_00.to_d],
-        BigDecimal("Infinity") => [0.15, 10_245_00.to_d]
+        148_289_00.to_d => [LOWEST_RATE, 0.0.to_d],
```
[the remark this answers is not in the document]
Comment (mark-h-humi) [reply lib/taxman/taxman2024/ab/t4.rb:11]: d'oh

**Review by markjconlon-humi — APPROVED** [review]

**Pull request merged**